go.mod

```diff
@@ -1,20 +1,22 @@
 module github.com/fullstorydev/grpchan
 
-go 1.18
+go 1.22.7
+
+replace google.golang.org/grpc => github.com/dfawley/grpc-go v1.49.0-dev.0.20241101162700-fcd45dd4befc
```

Comment:
Ohh. you're just trying out the patch.  Gotcha.

Reply:
Yes. However, this version of grpc-go will not be backwards compatible. So users that update to the. new version of grpc-go will _need_ a new version of this repo, and the compatibility fix requires in this repo means it must depend on the newer grpc-go (and thus inherit this constraint).

So it will be necessary for this to be the minimum once these changes are finalized, solely because that's the new minimum for grpc-go.

FWIW, at Buf we also support 3 versions. And I'm a bit surprised that grpc-go has already jumped from 1.21 since the Google Cloud Go support policy is also to support 3 versions: https://github.com/googleapis/google-cloud-go?tab=readme-ov-file#supported-versions. (Though it is changing in Jan to only support two.) This means that the Google Cloud Go SDKs can't update to the latest grpc-go until January.

@dfawley, is this something you are already aware of?

Reply:
Unfortunately, yes, we're aware of it.  Now that the Go support version declarations in `go.mod` are viral, and because most of the community only follows a 2 version support policy, we're effectively forced to adopt it as well -- or update our dependencies behind their release schedules, thereby risking delaying security patches as a result.

I chatted with the cloud API team about this (cc @codyoss) and they said it was OK to go ahead with this.